### PR TITLE
chore: deposit tests with real BLS signatures [BS-5016]

### PR DESCRIPTION
**Pull request by zajck**
## Description

Tests were mocking BLS signature, since EIP-2537 precompiles weren't enabled in Foundry by default.
They are now, so this PR adds a signing utility that creates a valid BLS signature.

-  Added new util `BLSSigner.sol`, which signs the deposit data. It reuses `BLS12_381.sol` where possible. Note that it needs to call `BLS12_381.hashToG2` through a self-staticcall trampoline, since calling it directly multiple times in a single call results in a dirty scratchpad and wrong hash.
- `ConsensusLayerDepositManagerAttestation.t.sol` is updated to use actual signatures. New test were added, making sure wrong BLS signatures fail.
- `River.1.t.sol` and `AccountingHarnessBase.sol` still use mock `verifyBLSDeposit` call since the checks in those tests do not depend on valid BLS signatures. Only the comment before mocking was updated.

## Notice

- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/liquid-collective/liquid-collective-protocol/pulls) for the same update/change?
- [ ] Have you assigned this PR to yourself?
- [ ] Have you added at least 1 reviewer?
- [ ] Have you updated the official documentation?
- [ ] Have you added sufficient documentation in your code?
- [ ] Have you added relevant tests to the official test suite?

## Pull Request Type

- [ ] 💫 New Feature (Breaking Change)
- [ ] 💫 New Feature (Non-breaking Change)
- [ ] 🛠️ Bug fix (Non-breaking Change: Fixes an issue)
- [ ] 🕹️ Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)

## Testing

- [ ] Have you tested this code with the official test suite?
- [ ] Have you tested this code manually?

### CODE REVIEW

File: contracts/test/utils/BLSSignerValidation/BLSFlagEdgeCases.t.sol [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+        returns (bytes32)
+    {
+        return BLS12_381.depositMessageSigningRoot(pubkey, amount, wc, domain);
+    }
+}
+
+/**
+ * @title BLSFlagEdgeCasesTest
+ * @notice Solidity counterparts to the `BLS flag edge cases` block of noble-curves'
+ *         `test/bls12-381.test.ts`, plus its infinity-point and non-canonical-limb cases.
```
Comment (juliaaschmidt) [inline contracts/test/utils/BLSSignerValidation/BLSFlagEdgeCases.t.sol:53]: Sounds like this was taken from a typescript project? l.53-l.68 if we adapt work from other projects, we should cite them on the top of each file.

File: contracts/test/utils/BLSSignerValidation/BLSVectors.sol
Comment (juliaaschmidt) [inline contracts/test/utils/BLSSignerValidation/BLSVectors.sol:1]: if this is auto-generated file we should move all the generated files into a `generated/` path or some other rule to make it clear.

File: scripts/bls-gen-vectors.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env node
+//
+// Regenerates contracts/test/utils/BLSSignerValidation/BLSVectors.sol.
+//
+// Two sources, deliberately:
+//   1. The `sign` handler of ethereum/bls12-381-tests (CC0-1.0) — official, authoritative, and
```
Comment (juliaaschmidt) [inline scripts/bls-gen-vectors.js:6]: The licensing is confusing. What licensing is this file? We put  BUSL-1.1 on the other new files but this one mentions CC0-1.0 fixture data.

File: contracts/test/components/ConsensusLayerDepositManagerAttestation.t.sol [hunk 3]
```diff
@@ -934,6 +925,192 @@ contract ConsensusLayerDepositManagerAttestationTest is Test {
...
+    //
+    // The batch is always tampered with *before* _prepareDeposit, because the buffer id commits
+    // to the batch contents: mutating a deposit afterwards would trip BufferIdMismatch first and
+    // the BLS path would never be reached.
+    // -----------------------------------------------------------------------
+
+    /// @dev A signature that is a valid BLS signature, just by the wrong key. Both the signature
+    ///      and its Y coordinate come from key B, so the point is well-formed and on-curve and the
+    ///      pairing check runs to a clean `false` rather than failing the precompile.
+    function testRevert_blsRejectsSignatureFromDifferentKey() public {
```
Comment (juliaaschmidt) [inline contracts/test/components/ConsensusLayerDepositManagerAttestation.t.sol:943]: Finally some great tests in this file 🥳

File: package.json [hunk 4]
```diff
@@ -10,9 +10,12 @@
     "compile": "hardhat compile",
     "hh": "hardhat",
     "uml": "sol2uml ./contracts/src -o ./docs/components.svg",
-    "link_contracts": "cd node_modules && rm -rf openzeppelin-contracts && ln -s ../lib/openzeppelin-contracts && rm -rf openzeppelin-contracts-upgradeable && ln -s ../lib/openzeppelin-contracts-upgradeable"
+    "link_contracts": "cd node_modules && rm -rf openzeppelin-contracts && ln -s ../lib/openzeppelin-contracts && rm -rf openzeppelin-contracts-upgradeable && ln -s ../lib/openzeppelin-contracts-upgradeable",
+    "bls:vectors": "node scripts/bls-gen-vectors.js",
+    "bls:crosscheck": "node scripts/bls-crosscheck.js"
   },
   "dependencies": {
+    "@noble/curves": "1.4.2",
```
Comment (juliaaschmidt) [inline package.json:18]: technically a dev dependency only, also why the old version? https://www.npmjs.com/package/@noble/curves 2.4.0 is the latest

**Review by iamsahu — DISMISSED** [review]
LGTM! 🚀 

It would be great to have all references featured prominently at the top of the files wherever needed.

**Review by juliaaschmidt — DISMISSED** [review]

File: package.json (hunk 4 above)
Comment (juliaaschmidt) [reply package.json:18]: just can hear renovatebot yelling at us already

File: scripts/bls-gen-vectors.js [hunk 5]
```diff
@@ -1,6 +1,10 @@
 #!/usr/bin/env node
 //
-// Regenerates contracts/test/utils/BLSSignerValidation/BLSVectors.sol.
+// This script is licensed like the rest of the repo (BUSL-1.1, see LICENSE). It downloads and
+// embeds fixture data from ethereum/bls12-381-tests, which that project licenses CC0-1.0 — that
```
Comment (juliaaschmidt) [inline scripts/bls-gen-vectors.js:4]: 🔥

**Review by juliaaschmidt — APPROVED** [review]
LGTM

**Review by iamsahu — APPROVED** [review]
LGTM!
